app/src/main/java/io/github/dkter/aaaaa/AaaaaInputMethodService.kt

```diff
@@ -47,6 +60,14 @@ class AaaaaInputMethodService:
         inputChar('a')
     }
 
+    override fun onLongA() {
+        longPressThread = newLongPressThread()
+    }
+
+    override fun onReleaseA() {
+        longPressThread?.interrupt()
```

Comment:
An edge case here is that the thread keeps running e.g. when you hold down the `a` key and press Back or Home. This can be stopped by just pulling up the keyboard again and pressing `a` again but it would be nice to also interrupt the thread when the keyboard goes out of focus.

(if instead of pressing `a` again, you long-press it, it doesn't stop and just accelerates the `a`s and there doesn't seem to be any way of killing it short of force-stopping the app o:)

Reply:
Thanks for letting me know, I'll fix this. I can probably figure this out myself, but if you happen to know, can you tell me how to listen to the event of the keyboard closing?

Reply:
you could try [`InputMethodService.onFinishInputView`](https://developer.android.com/reference/android/inputmethodservice/InputMethodService#onFinishInputView(boolean)) or something similar

Reply:
Thank you!

Reply:
Fixed in da2a1cf.